reactor-core/src/main/java/reactor/core/publisher/OptimisticEmitFailureHandler.java

```diff
@@ -0,0 +1,44 @@
+package reactor.core.publisher;
+
+import java.time.Duration;
+
+/**
+ * @author Animesh Chaturvedi
+ * This implementation deals with the transient failure in case of emit from
+ * sink because of competing threads. Busy looping is used as an approach instead
+ * of {@link Sinks.EmitFailureHandler#FAIL_FAST} with specified duration
+ */
+public class OptimisticEmitFailureHandler implements Sinks.EmitFailureHandler {
+
+	/**
+	 * Duration for busy looping. Be careful in providing the duration as the
+	 * construction, emission and handling for the first try will take more than the
+	 * duration provided which defeats the purpose. Provide duration > 100ms
+	 */
+    private final long duration;
+
+	/**
+	 * Static factory method to get Emit failure handler which handles the busy looping
+	 * for transient failure case.
+	 * @param duration Duration specified for busy looping
+	 * @return an instance of {@link OptimisticEmitFailureHandler}
+	 */
```

Comment:
This factory method should be moved to the `EmitFailureHandler` class and return an `EmitFailureHandler` (as `OptimisticEmitFailureHandler` would be turned into a package-private class).

Here is my suggested javadoc for the resulting static factory method, which syntheses all the pieces of information spread in this class:

```java
/**
 * Create an {@link EmitFailureHandler} which will busy loop in case of concurrent use
 * of the sink ({@link EmitResult#FAIL_NON_SERIALIZED}, up to a deadline.
 * The deadline is computed immediately from the current time (construction time) 
 * + provided {@link Duration}.
 * <p>
 * As a result there will always be some delay between this computation and the actual first
 * use of the handler (at a minimum, the time it takes for the first sink emission attempt).
 * Consider this when choosing the {@link Duration}, and probably prefer something above 100ms.
 *
 * @param duration {@link Duration} for the deadline
 * @return an optimistic and bounded busy-looping {@link EmitFailureHandler}
 */
```